web/src/components/Calendar/Eventcard.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { ReactElement } from 'react'
+
+import Date from './Date'
+import Description from './Description'
+
+interface IProps {
+  children: ReactElement | Array<ReactElement>
+}
+
+interface IEventcard extends React.FC<IProps> {
+  Date: typeof Date
+  Description: typeof Description
+}
+
+const Eventcard: IEventcard = props => (
+  <div className='calendar__eventcard'>
+    {props.children}
+  </div>
+)
+
+Eventcard.Date = Date
+Eventcard.Description = Description
```

Comment:
Γιατι να δινουμε children αντι για να μην δινουμε κατι fixed εδώ; Αφου ένα Date και ενα Description δεν θα εχει κάθε Event;
Άρα μπορείς απευθείας να τα γράψεις εσύ μέσα στον κώδικα.

Reply:
Ο λόγος που έβαλα children και όχι κατι fixed είναι ότι το Eventcard δεν αντιπροσωπεύει κάθε event. Αντιπροσωπεύει όλα τα events που γίνονται σε συγκεκριμένη ημερομηνία. Αυτό σημαίνει ότι θα υπάρχει σίγουρα ένα date αλλά τα events που θα γίνονται σε αυτή την ημερομηνία (Descriptions) μπορεί να είναι παραπάνω από ένα.

Reply:
οκ. Αλλά το Description σαν ονομα δεν λεει τιποτα. Μπορεις να το πεις απλα `Event`, ή `EventItem`. Γενικά όταν λες "περιγραφή" αναφέρεσαι σε μια περιγραφή άνα αντικείμενο.

Επίσης, εφόσων το Date είναι σίγουρο ότι θα υπάρχει και πρέπει να υπάρχει, μπορείς απλα να το αντικαταστήσεις με `property`.
Τα `Events` επίσης, γιατί είναι δεδομένα που έρχονται από database και όχι φτιαγμένα on the go στις σελίδες και βολεύει περισσότερο να τα παίρνει μια και καλή και να αναλαμβάνει μόνο του το render. :)

Reply:
```tsx
interface IProps {
  events: IEvent[] // Υποθετοντας οτι υπαρχει interface αντιστοιχο που το αντιπροσωπευει
}

//*************************

// REMOVE
// interface IEventcard extends React.FC<IProps> {
//   Date: typeof Date
//   Events: typeof EventItem
// }

//*********************
const EventCard: React.FC<IProps>

//*********************

// REMOVE Eventcard.Date = Date
// REMOVE Eventcard.Description = Description
```

Reply:
Δίνεις επίσης πάνω από δυο φορές την ημερομηνία.
Μια στο `Date`, και αλλή μια τουλάχιστον για κάθε event που έχει μέσα το property "date" που λέει ούτε σι άλλος πότε διοργανώνεται.